[Bug] WelchPSD, WelchCohere, signal.windows.hanning removed with scipy 1.9.0

**Describe the bug**
With release of scipy 1.9.0, `signal.windows.hanning` was removed.
See release notes: https://github.com/scipy/scipy/releases/tag/v1.9.0.

Deprecated since: 1.1.0 see: https://github.com/scipy/scipy/pull/8148

**To Reproduce**
1. install elephant
2. install scipy v1.9.0
3. run unit tests for spectral

**Environment**
 - OS (e.g., Linux): Ubuntu 20.04.4 LTS
 - How you installed elephant (`conda`, `pip`, source): source
 - Python version: 3.8
 - `neo` python package version: 0.10.2
 - `elephant` python package version: 0.12.0b1
 - `scipy`  python package version: 1.9.0

**Log**

```
elephant/test/test_spectral.py:48 (WelchPSDTestCase.test_welch_psd_behavior)
window = 'hanning', Nx = 1000, fftbins = True

    def get_window(window, Nx, fftbins=True):
        """
        Return a window of a given length and type.
    
        Parameters
        ----------
        window : string, float, or tuple
            The type of window to create. See below for more details.
        Nx : int
            The number of samples in the window.
        fftbins : bool, optional
            If True (default), create a "periodic" window, ready to use with
            `ifftshift` and be multiplied by the result of an FFT (see also
            :func:`~scipy.fft.fftfreq`).
            If False, create a "symmetric" window, for use in filter design.
    
        Returns
        -------
        get_window : ndarray
            Returns a window of length `Nx` and type `window`
    
        Notes
        -----
        Window types:
    
        - `~scipy.signal.windows.boxcar`
        - `~scipy.signal.windows.triang`
        - `~scipy.signal.windows.blackman`
        - `~scipy.signal.windows.hamming`
        - `~scipy.signal.windows.hann`
        - `~scipy.signal.windows.bartlett`
        - `~scipy.signal.windows.flattop`
        - `~scipy.signal.windows.parzen`
        - `~scipy.signal.windows.bohman`
        - `~scipy.signal.windows.blackmanharris`
        - `~scipy.signal.windows.nuttall`
        - `~scipy.signal.windows.barthann`
        - `~scipy.signal.windows.cosine`
        - `~scipy.signal.windows.exponential`
        - `~scipy.signal.windows.tukey`
        - `~scipy.signal.windows.taylor`
        - `~scipy.signal.windows.kaiser` (needs beta)
        - `~scipy.signal.windows.kaiser_bessel_derived` (needs beta)
        - `~scipy.signal.windows.gaussian` (needs standard deviation)
        - `~scipy.signal.windows.general_cosine` (needs weighting coefficients)
        - `~scipy.signal.windows.general_gaussian` (needs power, width)
        - `~scipy.signal.windows.general_hamming` (needs window coefficient)
        - `~scipy.signal.windows.dpss` (needs normalized half-bandwidth)
        - `~scipy.signal.windows.chebwin` (needs attenuation)
    
    
        If the window requires no parameters, then `window` can be a string.
    
        If the window requires parameters, then `window` must be a tuple
        with the first argument the string name of the window, and the next
        arguments the needed parameters.
    
        If `window` is a floating point number, it is interpreted as the beta
        parameter of the `~scipy.signal.windows.kaiser` window.
    
        Each of the window types listed above is also the name of
        a function that can be called directly to create a window of
        that type.
    
        Examples
        --------
        >>> from scipy import signal
        >>> signal.get_window('triang', 7)
        array([ 0.125,  0.375,  0.625,  0.875,  0.875,  0.625,  0.375])
        >>> signal.get_window(('kaiser', 4.0), 9)
        array([ 0.08848053,  0.29425961,  0.56437221,  0.82160913,  0.97885093,
                0.97885093,  0.82160913,  0.56437221,  0.29425961])
        >>> signal.get_window(('exponential', None, 1.), 9)
        array([ 0.011109  ,  0.03019738,  0.082085  ,  0.22313016,  0.60653066,
                0.60653066,  0.22313016,  0.082085  ,  0.03019738])
        >>> signal.get_window(4.0, 9)
        array([ 0.08848053,  0.29425961,  0.56437221,  0.82160913,  0.97885093,
                0.97885093,  0.82160913,  0.56437221,  0.29425961])
    
        """
        sym = not fftbins
        try:
>           beta = float(window)
E           ValueError: could not convert string to float: 'hanning'

../../venv/lib/python3.8/site-packages/scipy/signal/windows/_windows.py:2214: ValueError

During handling of the above exception, another exception occurred:

window = 'hanning', Nx = 1000, fftbins = True

    def get_window(window, Nx, fftbins=True):
        """
        Return a window of a given length and type.
    
        Parameters
        ----------
        window : string, float, or tuple
            The type of window to create. See below for more details.
        Nx : int
            The number of samples in the window.
        fftbins : bool, optional
            If True (default), create a "periodic" window, ready to use with
            `ifftshift` and be multiplied by the result of an FFT (see also
            :func:`~scipy.fft.fftfreq`).
            If False, create a "symmetric" window, for use in filter design.
    
        Returns
        -------
        get_window : ndarray
            Returns a window of length `Nx` and type `window`
    
        Notes
        -----
        Window types:
    
        - `~scipy.signal.windows.boxcar`
        - `~scipy.signal.windows.triang`
        - `~scipy.signal.windows.blackman`
        - `~scipy.signal.windows.hamming`
        - `~scipy.signal.windows.hann`
        - `~scipy.signal.windows.bartlett`
        - `~scipy.signal.windows.flattop`
        - `~scipy.signal.windows.parzen`
        - `~scipy.signal.windows.bohman`
        - `~scipy.signal.windows.blackmanharris`
        - `~scipy.signal.windows.nuttall`
        - `~scipy.signal.windows.barthann`
        - `~scipy.signal.windows.cosine`
        - `~scipy.signal.windows.exponential`
        - `~scipy.signal.windows.tukey`
        - `~scipy.signal.windows.taylor`
        - `~scipy.signal.windows.kaiser` (needs beta)
        - `~scipy.signal.windows.kaiser_bessel_derived` (needs beta)
        - `~scipy.signal.windows.gaussian` (needs standard deviation)
        - `~scipy.signal.windows.general_cosine` (needs weighting coefficients)
        - `~scipy.signal.windows.general_gaussian` (needs power, width)
        - `~scipy.signal.windows.general_hamming` (needs window coefficient)
        - `~scipy.signal.windows.dpss` (needs normalized half-bandwidth)
        - `~scipy.signal.windows.chebwin` (needs attenuation)
    
    
        If the window requires no parameters, then `window` can be a string.
    
        If the window requires parameters, then `window` must be a tuple
        with the first argument the string name of the window, and the next
        arguments the needed parameters.
    
        If `window` is a floating point number, it is interpreted as the beta
        parameter of the `~scipy.signal.windows.kaiser` window.
    
        Each of the window types listed above is also the name of
        a function that can be called directly to create a window of
        that type.
    
        Examples
        --------
        >>> from scipy import signal
        >>> signal.get_window('triang', 7)
        array([ 0.125,  0.375,  0.625,  0.875,  0.875,  0.625,  0.375])
        >>> signal.get_window(('kaiser', 4.0), 9)
        array([ 0.08848053,  0.29425961,  0.56437221,  0.82160913,  0.97885093,
                0.97885093,  0.82160913,  0.56437221,  0.29425961])
        >>> signal.get_window(('exponential', None, 1.), 9)
        array([ 0.011109  ,  0.03019738,  0.082085  ,  0.22313016,  0.60653066,
                0.60653066,  0.22313016,  0.082085  ,  0.03019738])
        >>> signal.get_window(4.0, 9)
        array([ 0.08848053,  0.29425961,  0.56437221,  0.82160913,  0.97885093,
                0.97885093,  0.82160913,  0.56437221,  0.29425961])
    
        """
        sym = not fftbins
        try:
            beta = float(window)
        except (TypeError, ValueError) as e:
            args = ()
            if isinstance(window, tuple):
                winstr = window[0]
                if len(window) > 1:
                    args = window[1:]
            elif isinstance(window, str):
                if window in _needs_param:
                    raise ValueError("The '" + window + "' window needs one or "
                                     "more parameters -- pass a tuple.") from e
                else:
                    winstr = window
            else:
                raise ValueError("%s as window type is not supported." %
                                 str(type(window))) from e
    
            try:
>               winfunc = _win_equiv[winstr]
E               KeyError: 'hanning'

../../venv/lib/python3.8/site-packages/scipy/signal/windows/_windows.py:2232: KeyError

The above exception was the direct cause of the following exception:

self = <elephant.test.test_spectral.WelchPSDTestCase testMethod=test_welch_psd_behavior>

    def test_welch_psd_behavior(self):
        # generate data by adding white noise and a sinusoid
        data_length = 5000
        sampling_period = 0.001
        signal_freq = 100.0
        noise = np.random.normal(size=data_length)
        signal = [np.sin(2 * np.pi * signal_freq * t)
                  for t in np.arange(0, data_length * sampling_period,
                                     sampling_period)]
        data = n.AnalogSignal(np.array(signal + noise),
                              sampling_period=sampling_period * pq.s,
                              units='mV')
    
        # consistency between different ways of specifying segment length
>       freqs1, psd1 = elephant.spectral.welch_psd(
            data, len_segment=data_length // 5, overlap=0)

test_spectral.py:63: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../utils.py:80: in wrapper
    return func(*args, **kwargs)
../spectral.py:260: in welch_psd
    freqs, psd = scipy.signal.welch(**params)
../../venv/lib/python3.8/site-packages/scipy/signal/_spectral_py.py:446: in welch
    freqs, Pxx = csd(x, x, fs=fs, window=window, nperseg=nperseg,
../../venv/lib/python3.8/site-packages/scipy/signal/_spectral_py.py:580: in csd
    freqs, _, Pxy = _spectral_helper(x, y, fs, window, nperseg, noverlap, nfft,
../../venv/lib/python3.8/site-packages/scipy/signal/_spectral_py.py:1780: in _spectral_helper
    win, nperseg = _triage_segments(window, nperseg, input_length=x.shape[-1])
../../venv/lib/python3.8/site-packages/scipy/signal/_spectral_py.py:2003: in _triage_segments
    win = get_window(window, nperseg)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

window = 'hanning', Nx = 1000, fftbins = True

    def get_window(window, Nx, fftbins=True):
        """
        Return a window of a given length and type.
    
        Parameters
        ----------
        window : string, float, or tuple
            The type of window to create. See below for more details.
        Nx : int
            The number of samples in the window.
        fftbins : bool, optional
            If True (default), create a "periodic" window, ready to use with
            `ifftshift` and be multiplied by the result of an FFT (see also
            :func:`~scipy.fft.fftfreq`).
            If False, create a "symmetric" window, for use in filter design.
    
        Returns
        -------
        get_window : ndarray
            Returns a window of length `Nx` and type `window`
    
        Notes
        -----
        Window types:
    
        - `~scipy.signal.windows.boxcar`
        - `~scipy.signal.windows.triang`
        - `~scipy.signal.windows.blackman`
        - `~scipy.signal.windows.hamming`
        - `~scipy.signal.windows.hann`
        - `~scipy.signal.windows.bartlett`
        - `~scipy.signal.windows.flattop`
        - `~scipy.signal.windows.parzen`
        - `~scipy.signal.windows.bohman`
        - `~scipy.signal.windows.blackmanharris`
        - `~scipy.signal.windows.nuttall`
        - `~scipy.signal.windows.barthann`
        - `~scipy.signal.windows.cosine`
        - `~scipy.signal.windows.exponential`
        - `~scipy.signal.windows.tukey`
        - `~scipy.signal.windows.taylor`
        - `~scipy.signal.windows.kaiser` (needs beta)
        - `~scipy.signal.windows.kaiser_bessel_derived` (needs beta)
        - `~scipy.signal.windows.gaussian` (needs standard deviation)
        - `~scipy.signal.windows.general_cosine` (needs weighting coefficients)
        - `~scipy.signal.windows.general_gaussian` (needs power, width)
        - `~scipy.signal.windows.general_hamming` (needs window coefficient)
        - `~scipy.signal.windows.dpss` (needs normalized half-bandwidth)
        - `~scipy.signal.windows.chebwin` (needs attenuation)
    
    
        If the window requires no parameters, then `window` can be a string.
    
        If the window requires parameters, then `window` must be a tuple
        with the first argument the string name of the window, and the next
        arguments the needed parameters.
    
        If `window` is a floating point number, it is interpreted as the beta
        parameter of the `~scipy.signal.windows.kaiser` window.
    
        Each of the window types listed above is also the name of
        a function that can be called directly to create a window of
        that type.
    
        Examples
        --------
        >>> from scipy import signal
        >>> signal.get_window('triang', 7)
        array([ 0.125,  0.375,  0.625,  0.875,  0.875,  0.625,  0.375])
        >>> signal.get_window(('kaiser', 4.0), 9)
        array([ 0.08848053,  0.29425961,  0.56437221,  0.82160913,  0.97885093,
                0.97885093,  0.82160913,  0.56437221,  0.29425961])
        >>> signal.get_window(('exponential', None, 1.), 9)
        array([ 0.011109  ,  0.03019738,  0.082085  ,  0.22313016,  0.60653066,
                0.60653066,  0.22313016,  0.082085  ,  0.03019738])
        >>> signal.get_window(4.0, 9)
        array([ 0.08848053,  0.29425961,  0.56437221,  0.82160913,  0.97885093,
                0.97885093,  0.82160913,  0.56437221,  0.29425961])
    
        """
        sym = not fftbins
        try:
            beta = float(window)
        except (TypeError, ValueError) as e:
            args = ()
            if isinstance(window, tuple):
                winstr = window[0]
                if len(window) > 1:
                    args = window[1:]
            elif isinstance(window, str):
                if window in _needs_param:
                    raise ValueError("The '" + window + "' window needs one or "
                                     "more parameters -- pass a tuple.") from e
                else:
                    winstr = window
            else:
                raise ValueError("%s as window type is not supported." %
                                 str(type(window))) from e
    
            try:
                winfunc = _win_equiv[winstr]
            except KeyError as e:
>               raise ValueError("Unknown window type.") from e
E               ValueError: Unknown window type.

../../venv/lib/python3.8/site-packages/scipy/signal/windows/_windows.py:2234: ValueError

```

--- a/elephant/spectral.py
+++ b/elephant/spectral.py
@@ -13,10 +13,9 @@ spectrum).
 :license: Modified BSD, see LICENSE.txt for details.
 """
 
-from __future__ import division, print_function, unicode_literals
+import warnings
 
 import neo
-import warnings
 import numpy as np
 import quantities as pq
 import scipy.signal

--- a/elephant/spectral.py
+++ b/elephant/spectral.py
@@ -32,7 +32,7 @@ __all__ = [
 @deprecated_alias(num_seg='n_segments', len_seg='len_segment',
                   freq_res='frequency_resolution')
 def welch_psd(signal, n_segments=8, len_segment=None,
-              frequency_resolution=None, overlap=0.5, fs=1.0, window='hanning',
+              frequency_resolution=None, overlap=0.5, fs=1.0, window='hann',
               nfft=None, detrend='constant', return_onesided=True,
               scaling='density', axis=-1):
     """
@@ -91,7 +91,7 @@ def welch_psd(signal, n_segments=8, len_segment=None,
     window : str or tuple or np.ndarray, optional
         Desired window to use.
         See Notes [2].
-        Default: 'hanning'
+        Default: 'hann'
     nfft : int, optional
         Length of the FFT used.
         See Notes [2].
@@ -520,7 +520,7 @@ def multitaper_psd(signal, n_segments=1, len_segment=None,
                   len_seg='len_segment', freq_res='frequency_resolution')
 def welch_coherence(signal_i, signal_j, n_segments=8, len_segment=None,
                     frequency_resolution=None, overlap=0.5, fs=1.0,
-                    window='hanning', nfft=None, detrend='constant',
+                    window='hann', nfft=None, detrend='constant',
                     scaling='density', axis=-1):
     r"""
     Estimates coherence between a given pair of analog signals.
@@ -574,7 +574,7 @@ def welch_coherence(signal_i, signal_j, n_segments=8, len_segment=None,
     window : str or tuple or np.ndarray, optional
         Desired window to use.
         See Notes [1].
-        Default: 'hanning'
+        Default: 'hann'
     nfft : int, optional
         Length of the FFT used.
         See Notes [1].

--- a/elephant/spectral.py
+++ b/elephant/spectral.py
@@ -195,7 +195,13 @@ def welch_psd(signal, n_segments=8, len_segment=None,
         3.14573483e-08, 6.82050475e-09, 1.18183354e-10]]) * mV**2/Hz
 
     """
-
+    # 'hanning' window was removed with release of scipy 1.9.0, it was
+    # deprecated since 1.1.0.
+    if window == 'hanning':
+        warnings.warn("'hanning' is deprecated and was removed from scipy "
+                      "with release 1.9.0. Please use 'hann' instead",
+                      DeprecationWarning)
+        window = 'hann'
     # initialize a parameter dict (to be given to scipy.signal.welch()) with
     # the parameters directly passed on to scipy.signal.welch()
     params = {'window': window, 'nfft': nfft,
@@ -670,6 +676,13 @@ def welch_coherence(signal_i, signal_j, n_segments=8, len_segment=None,
     """
 
     # TODO: code duplication with welch_psd()
+    # 'hanning' window was removed with release of scipy 1.9.0, it was
+    # deprecated since 1.1.0.
+    if window == 'hanning':
+        warnings.warn("'hanning' is deprecated and was removed from scipy "
+                      "with release 1.9.0. Please use 'hann' instead",
+                      DeprecationWarning)
+        window = 'hann'
 
     # initialize a parameter dict for scipy.signal.csd()
     params = {'window': window, 'nfft': nfft,

--- a/elephant/spectral.py
+++ b/elephant/spectral.py
@@ -329,7 +329,7 @@ def multitaper_psd(signal, n_segments=1, len_segment=None,
         Time bandwidth product
         Default: 4.0.
     num_tapers : int, optional
-        Number of tapers used in 1. to obtain estimate of PSD. By default
+        Number of tapers used in 1. to obtain estimate of PSD. By default,
         [2*nw] - 1 is chosen.
         Default: None.
     peak_resolution : pq.Quantity float, optional
@@ -607,7 +607,7 @@ def welch_coherence(signal_i, signal_j, n_segments=8, len_segment=None,
         Estimate of coherency between the input time series. For each
         frequency, coherency takes a value between 0 and 1, with 0 or 1
         representing no or perfect coherence, respectively.
-        When the input arrays `signal_i` and `signal_j` are multi-dimensional,
+        When the input arrays `signal_i` and `signal_j` are multidimensional,
         `coherency` is of the same shape as the inputs, and the frequency is
         indexed depending on the type of the input. If the input is
         `neo.AnalogSignal`, the first axis indexes frequency. Otherwise,

--- a/elephant/test/test_spectral.py
+++ b/elephant/test/test_spectral.py
@@ -139,7 +139,7 @@ class WelchPSDTestCase(unittest.TestCase):
         noise = np.random.normal(size=(num_channel, data_length))
         data_np = np.array(noise)
         # Since row-column order in AnalogSignal is different from the
-        # conventional one, `data_np` needs to be transposed when its used to
+        # conventional one, `data_np` needs to be transposed when it's used to
         # define an AnalogSignal
         data_neo = n.AnalogSignal(data_np.T,
                                   sampling_period=sampling_period * pq.s,
@@ -444,7 +444,7 @@ class WelchCohereTestCase(unittest.TestCase):
         x_np = np.array(np.random.normal(size=(num_channel, data_length)))
         y_np = np.array(np.random.normal(size=(num_channel, data_length)))
         # Since row-column order in AnalogSignal is different from the
-        # convention in NumPy/SciPy, `data_np` needs to be transposed when its
+        # convention in NumPy/SciPy, `data_np` needs to be transposed when it's
         # used to define an AnalogSignal
         x_neo = n.AnalogSignal(x_np.T, units='mV',
                                sampling_period=sampling_period * pq.s)

--- a/elephant/test/test_spectral.py
+++ b/elephant/test/test_spectral.py
@@ -46,6 +46,14 @@ class WelchPSDTestCase(unittest.TestCase):
         self.assertRaises(ValueError, elephant.spectral.welch_psd, data,
                           overlap=1.1)
 
+    def test_welch_psd_warnings(self):
+        # generate a dummy data
+        data = n.AnalogSignal(np.zeros(5000), sampling_period=0.001 * pq.s,
+                              units='mV')
+        # Test deprecation warning for 'hanning' window
+        self.assertWarns(DeprecationWarning, elephant.spectral.welch_psd,
+                         data, window='hanning')
+
     def test_welch_psd_behavior(self):
         # generate data by adding white noise and a sinusoid
         data_length = 5000
@@ -336,6 +344,16 @@ class WelchCohereTestCase(unittest.TestCase):
                           overlap=-1.0)
         self.assertRaises(ValueError, elephant.spectral.welch_coherence, x, y,
                           overlap=1.1)
+
+    def test_welch_cohere_warnings(self):
+        # generate a dummy data
+        x = n.AnalogSignal(np.zeros(5000), sampling_period=0.001 * pq.s,
+                           units='mV')
+        y = n.AnalogSignal(np.zeros(5000), sampling_period=0.001 * pq.s,
+                           units='mV')
+        # Test deprecation warning for 'hanning' window
+        self.assertWarns(DeprecationWarning, elephant.spectral.welch_coherence,
+                         x, y, window='hanning')
 
     def test_welch_cohere_behavior(self):
         # generate data by adding white noise and a sinusoid

